core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunction.java

```diff
@@ -49,6 +49,16 @@
      *            the value which decay function should take once the distance
      *            reaches this scale
      * */
-    public double processScale(double scale, double decay);
+    public abstract double processScale(double scale, double decay);
 
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
I forgot to impl hashcode.... will add. We need equals to be able to compare queries in tests when they contain functions.